aten/src/ATen/native/cuda/SoftMax.cu

```diff
@@ -1008,5 +1016,71 @@ Tensor masked_softmax_cuda(const Tensor& input, const Tensor& mask) {
     }
     return output;
 }
+
+Tensor masked_softmax_backward_cuda(
+    const Tensor& grad_,
+    const Tensor& output_,
+    int64_t dim_,
+    const Tensor& mask_) {
+  auto grad = grad_.contiguous();
+  auto output = output_.contiguous();
+  auto mask = mask_.is_contiguous() ? mask_ : mask_.contiguous();
+
+  grad = grad.dim() == 0 ? grad.view(1) : grad;
+  mask = mask.dim() == 0 ? mask.view(1) : mask;
+  output = output.dim() == 0 ? output.view(1) : output;
+  grad = grad * output;
+  int64_t dim = maybe_wrap_dim(dim_, grad.dim());
+
+  TORCH_CHECK(dim >=0 && dim < grad.dim(), "dim must be non-negative and less than input dimensions");
```

Comment:
you don't need this, `maybe_wrap_dim` will error out on out-of-range dim (that's why you are sending second argument to it)